tests/test_chebyshev.py

```diff
@@ -0,0 +1,135 @@
+
+
+class FgcmChebyshevTest(lsst.utils.tests.TestCase):
+    def setUp(self):
+        self.xSize = 2000
+        self.ySize = 4000
+        self.nStar = 1000
+
+        self.order = 2
+        self.pars = np.zeros((self.order + 1, self.order + 1))
+        self.pars[0, 0] = 1.0
+        self.pars[0, 1] = 3e-3
+        self.pars[0, 2] = 1e-5
+        self.pars[1, 0] = 5e-3
+        self.pars[1, 1] = 7e-6
+        self.pars[1, 2] = 0.0
+        self.pars[2, 0] = 3e-5
+        self.pars[2, 1] = 0.0
+        self.pars[2, 2] = 0.0
+
+        self.order2 = 1
+        self.pars2 = np.zeros((self.order2 + 1, self.order2 + 1))
+        self.pars2[0, 0] = 0.98
+        self.pars2[0, 1] = 5e-3
+        self.pars2[1, 0] = 2e-2
+
+    def test_chebyshev_evaluate(self):
+        # Set numpy seed for stability
+        np.random.seed(seed=1000)
+
+        xPos = self.xSize * np.random.rand(self.nStar)
+        yPos = self.ySize * np.random.rand(self.nStar)
+
+        bbox = lsst.geom.Box2I(lsst.geom.Point2I(0, 0),
+                               lsst.geom.Point2I(self.xSize - 1, self.ySize - 1))
+
+        # Compute the chebyshev values using the fgcm code
+        fgcmField = fgcm.fgcmUtilities.Cheb2dField(self.xSize, self.ySize,
+                                                   self.pars)
+        fgcmValues = fgcmField.evaluate(xPos, yPos)
+
+        # Compute the chebyshev values using the afw code
+        field = afwMath.ChebyshevBoundedField(bbox, self.pars)
+        fieldValues = field.evaluate(xPos, yPos)
+
+        self.assertFloatsAlmostEqual(fieldValues, fgcmValues, atol=1e-14)
+
+    def test_chebyshev_fit(self):
+        # Set numpy seed for stability
+        np.random.seed(seed=1000)
+
+        # Generate some points to fit
+        xPos = self.xSize * np.random.rand(self.nStar)
+        yPos = self.ySize * np.random.rand(self.nStar)
+        fgcmField = fgcm.fgcmUtilities.Cheb2dField(self.xSize, self.ySize,
+                                                   self.pars)
+        fgcmValues = fgcmField.evaluate(xPos, yPos)
+
+        # Fit the points using the fgcm code
+        fgcmField = fgcm.fgcmUtilities.Cheb2dField.fit(self.xSize, self.ySize,
+                                                       self.order, xPos, yPos,
+                                                       fgcmValues)
+
+        # Fit the points using the afw code
+        bbox = lsst.geom.Box2I(lsst.geom.Point2I(0, 0),
+                               lsst.geom.Point2I(self.xSize - 1, self.ySize - 1))
+
+        ctrl = afwMath.ChebyshevBoundedFieldControl()
+        ctrl.orderX = self.order
+        ctrl.orderY = self.order
+        ctrl.triangular = True
+        field = afwMath.ChebyshevBoundedField.fit(bbox, xPos, yPos,
+                                                  fgcmValues, ctrl)
+
+        # Compare the fit parameters
+        self.assertFloatsAlmostEqual(fgcmField.pars, field.getCoefficients(),
+                                     atol=1e-10)
```

Comment:
I assume that the tolerance is larger here because there are many more ways for fitters to return slightly different results. Are the objective function and fitting algorithm the same (should be yes for the former, maybe not for the latter)? If yes for both, you might want a smaller tolerance. Maybe it's worth comparing the values of the objective function too.